### ISSUE TITLE
[Hold for #38025] [$1000] Chat - Email id not shown in full available space & after sending full email iID shown for public accounts

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#35725]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.36
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Launch app
2. Tap in a report
3. Enter @ in compose box
4. Select an user from suggestions list
5. Note email id is not displayed occupying full space
6. Send the message

## Expected Result:
After selecting an email ID from the user suggestion list, it should be displayed in full within the available space. Once sent, the full email ID should not be displayed.

## Actual Result:
The selected email id from user suggestion list is not shown in full available space and after sending full email id is displayed.The selected email ID from the user suggestion list is not displayed in full within the available space. Additionally, after sending, the full email ID is still displayed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/b78b686a-1d83-4555-a1eb-daf122688972

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015976cfa3c5a4e481</li>
        <li>Upwork Job ID: 1753528802729029632</li>
        <li>Last Price Increase: 2024-03-07</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb
CC @quinthar

**Comment by MitchExpensify** [comment]
I am having trouble understanding this bug. Can you please rephrase the expected and actual results @kbecciv?

**Comment by Santhosh-Sellavel** [comment]
<img width="323" alt="Screenshot 2024-02-07 at 2 15 34 AM" src="https://github.com/Expensify/App/assets/85645967/3d98f478-64c3-4065-baa8-96854523691b">

@MitchExpensify 
Check the above screenshot taken from the issue description.

### Bug 1

In the composer box email is broken into multiple lines due to a long email, but still, some space on the right side is there.

The same email uses the full available space after sending it in the message area
<img width="294" alt="Screenshot 2024-02-07 at 2 18 06 AM" src="https://github.com/Expensify/App/assets/85645967/8be6333b-799c-473c-a6c8-a538d413d8f7">

### Bug 2
 > Additionally, after sending, the full email ID is still displayed
 
 I don't get this properly. But I guess 
 After selecting a user for mentioning from the suggested options, the message should have had the user name in the message instead of the email ID if the user name is available.
 
 Example `@Mitch instead mitch@expensify.com`
 
 Can you confirm @kbecciv ?

**Comment by kbecciv** [comment]
Yes, correct! @Santhosh-Sellavel

**Comment by Santhosh-Sellavel** [comment]
Bump @MitchExpensify

**Comment by Santhosh-Sellavel** [comment]
We are still waiting for a confirmation from @MitchExpensify to move forward and proposals as well.
Unassigning as am planning for OOO

**Comment by MitchExpensify** [comment]
Ok got it! Yes lets fix this. I think it could fit in #vip-vsb

**Comment by MitchExpensify** [comment]
Thanks for digging into the issue @Victor-Nyagudi ! Going to add this as LOW to vip vsb while we wait on proposals

**Comment by MitchExpensify** [comment]
Switching to Weekly as this is LOW

**Comment by MitchExpensify** [comment]
@shubham1206agra I notice this is labelled Internal, double checking if you think this can be made External or not

**Comment by MitchExpensify** [comment]
Highlighting for proposals here https://expensify.slack.com/archives/C01GTK53T8Q/p1710107159468519

**Comment by rojiphil** [comment]
@shubham1206agra I also could not reproduce the second part of the bug in main. But, I could reproduce the first part of the bug. Looking into this a little more closely.

**Comment by rojiphil** [comment]
> I think you misunderstood the problem here.

Oh Silly me! What a waste of my energy here 😓 . Hmmm. But this could be a potential improvement. Will post this in Slack and see if people like it.

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

E-mail id is not shown in full available space as shown in the screenshot below

![Screenshot 2024-03-12 at 10 58 00 AM](https://github.com/Expensify/App/assets/3004659/a48ef25e-1e0c-4f71-801f-e95295cbcdd7)

The second part of the issue (i.e. after sending email id) seems to be resolved in the latest main.

### What is the root cause of that problem?

[Here](https://github.com/Expensify/App/blob/b8590c966c7a7f7df8b6a1c8d8566cbab0fd572b/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx#L737) we set the styles for Composer using `textInputCompose` which sets the horizontal padding to `12` [here](https://github.com/Expensify/App/blob/b8590c966c7a7f7df8b6a1c8d8566cbab0fd572b/src/styles/index.ts#L1974). However, for smaller screens, real estate is very precious and the current value of horizontal padding of 12 waste away precious space which can be otherwise used for display of text. This seems to be the root cause of this problem.

### What changes do you think we should make in order to solve the problem?

Since there is already a padding of `10` set for the neighbouring `emoji`, we can reduce the horizontal padding set for Composer. Also, since the current vertical padding of `5` seems good enough for small screen devices, we can set the horizontal padding also to `5` and save some precious real estate for smaller devices.

For this, we can add a style like this [here](https://github.com/Expensify/App/blob/b8590c966c7a7f7df8b6a1c8d8566cbab0fd572b/src/styles/index.ts#L2006)

        smallTextInputComposeHorizontalPadding: {
            paddingHorizontal: variables.avatarChatSpacing - 7,
        },

And use this style for smaller screens like this for Composer [here](https://github.com/Expensify/App/blob/b8590c966c7a7f7df8b6a1c8d8566cbab0fd572b/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx#L737)

                    style={[styles.textInputCompose, isSmallScreenWidth && styles.smallTextInputComposeHorizontalPadding, isComposerFullSize ? styles.textInputFullCompose : styles.textInputCollapseCompose]}

Similar style change can also be made to `ReportActionItemMessageEdit`

Here is a comparison of how it will look before and after:

![Screenshot 2024-03-12 at 11 22 52 AM](https://github.com/Expensify/App/assets/3004659/b3716fec-0cec-4e1d-aea6-b799b44ecded)

### What alternative solutions did you explore? (Optional)

Alternatively, we can reduce the horizontal padding for all screen sizes if required. Also, if needed, we can further reduce the padding.

**Comment by rojiphil** [comment]
@shubham1206agra What do you think of this [proposal](https://github.com/Expensify/App/issues/35725#issuecomment-1990724232)? And I hope I am not way off here.

**Comment by rojiphil** [comment]
> I think the issue is related to line breaks happening at @ and .

@shubham1206agra That is not seen in the latest main. So, I am confused. 
Do you think there is still a pending issue to be resolved? Or it is all good here.

**Comment by rojiphil** [comment]
Sure. Will check on this.

**Comment by MitchExpensify** [comment]
Any update @rojiphil ?

**Comment by rojiphil** [comment]
Sorry for the delay in response here. 
@shubham1206agra I was able to reproduce the issue. And it looked like a nasty bug and I am nowhere closer to solving this.

**Comment by MitchExpensify** [comment]
Shared with expert agency https://expensify.slack.com/archives/C03UK30EA1Z/p1710810646724379

**Comment by Piotrfj** [comment]
Hi, Im Piotr from Callstack, I will start investigating this issue

**Comment by Piotrfj** [comment]
Update: found the source of the issue, now Im working on suitable solution that is not too big of walkoround

**Comment by MitchExpensify** [comment]
What's the latest @Piotrfj ?

**Comment by Piotrfj** [comment]
@shubham1206agra from what I see it's just native text wrapping behavior. There is no setting of that attribute that would change it exactly to what we want. Been trying to look for alternative solution but no luck for now. Right now I'm working on the second part of the issue and will consult the first one with others to find out if I'm missing something or not. That doesn't seem like to be connected to the live markdown though.

**Comment by Piotrfj** [comment]
Update: I have tested if the EmailUtils (mentioned here: https://github.com/Expensify/App/issues/35725#issuecomment-1991004933) are running, but they are not and it looks like all the breaking is due to android native behavior.

I have a question about the second part of the issue (mentioned here: https://github.com/Expensify/App/issues/35725#issuecomment-1930735370). I not sure if current behavior is a bug to be honest, I have followed the path of the values and everything seems to work as intended.
Do I understand correctly that we would want to change the behavior of what is filled in the input after choosing something from the mention tooltip to a nickname if such is available?

**Comment by MitchExpensify** [comment]
By "nickname" do you mean "Display Name"? @Piotrfj ?

**Comment by MitchExpensify** [comment]
> Do I understand correctly that we would want to change the behavior of what is filled in the input after choosing something from the mention tooltip to a nickname if such is available?

I think we always want to show the display name, is that what you are wondering?

**Comment by Piotrfj** [comment]
Current status for assignment change:

For the text breaking:
I've checked many settings but none of them worked as desired. I've found a function called `prefixMailSeparatorsWithBreakOpportunities` in EmailUtils.ts file, but it does not seem to run in this case. My current thought is that the issue is due to some native behavior of android and I could not get to the source of it, so I don't know if it's fixable.

For the display name part:
In file SuggestionMention.tsx at line 96
<img width="840" alt="image" src="https://github.com/Expensify/App/assets/41384305/3f0012bb-5fed-49a2-8b00-1cc8404b568f">
for both arguments `login` field is passed, as first argument the `text` field should be passed.
However after doing so the mention is no longer clickable and I haven't been able to find solution for that.

**Comment by MitchExpensify** [comment]
If our best thesis is that this is caused by native Andoird behavior I am tempted to close this. Its a minor issue that I'm not sure is worth digging deeper on at this point

**Comment by MitchExpensify** [comment]
Thoughts @Piotrfj ?

**Comment by koko57** [comment]
@MitchExpensify Piotr was reassigned to another project in Callstack. Someone from our team will take this issue over and will see if it's worth fixing

**Comment by MitchExpensify** [comment]
Ok got it!

**Comment by kubabutkiewicz** [comment]
Hi, I'm Jakub from Callstack - expert contributor group - I would like to take over this issue. Will go through convo here and will do my research and will leave my thoughts 😄

**Comment by MitchExpensify** [comment]
@koko57 any update on a new set of eyes on this one?

**Comment by burczu** [comment]
@MitchExpensify I'm taking a look at this one as Callstack engineer - please assign it to me. Thank you.

**Comment by MitchExpensify** [comment]
Done! Thanks @burczu

**Comment by MitchExpensify** [comment]
What is the latest here @burczu ? Thanks

**Comment by koko57** [comment]
@MitchExpensify - @burczu is OOO till 05/05. He's still working on a proper solution for this issue.

**Comment by MitchExpensify** [comment]
@burczu how is the solution coming along for this? Thanks

**Comment by burczu** [comment]
@MitchExpensify Hey! I'm back from ooo - sorry for not leaving an information about it here, I've completely forgotten to do it. I'm still involved in the Guided Setup Stage 2 project (there is one last PR to merge I'm taking care of) but I'll be also working on this issue this week.

Till now, I've confirmed what @Piotrfj found and described in his comment: https://github.com/Expensify/App/issues/35725#issuecomment-2052402367 - I think line breaking issue will be hard to fix and is minor, so i suggest leaving it as is. In terms of display name issue - I was looking for the solution for making it clickable after changing the format but didn't find the solution yet, so I'll continue an investigation.

**Comment by burczu** [comment]
Not overdue - still investigating.

**Comment by burczu** [comment]
Here's my update for the second part of the issue:

Mention highlighting is done in two places: for the chat message list it is done in the `MentionUserRenderer` component, for composer it must be done inside the `react-native-live-markdown` library.

For now I've managed to show display name instead of login only in the [MentionUserRenderer](https://github.com/Expensify/App/blob/main/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx) component, here's the result:

https://github.com/Expensify/App/assets/7771498/73b3e86c-8ab6-4abb-8687-373460ac4dd2

As you can see in the above video, the mention is not highlighted in the composer, and it shows a login as a mention - from what I've found, to fix it we need to change some native code of the `react-native-live-markdown`. I've tried to find the code responsible for this, but with no luck yet.

**Comment by burczu** [comment]
@shubham1206agra Thanks for pointing this out. Should we put this issue on hold as the one you've mentioned is also on hold now?

**Comment by MitchExpensify** [comment]
Placed on hold for https://github.com/Expensify/App/issues/38025

**Comment by MitchExpensify** [comment]
On hold for https://github.com/Expensify/App/issues/38025

**Comment by MitchExpensify** [comment]
On hold for https://github.com/Expensify/App/issues/38025

**Comment by MitchExpensify** [comment]
On hold for https://github.com/Expensify/App/issues/38025

**Comment by MitchExpensify** [comment]
On hold for https://github.com/Expensify/App/issues/38025

**Comment by MitchExpensify** [comment]
On hold for https://github.com/Expensify/App/issues/38025

**Comment by MitchExpensify** [comment]
Reassigning while I'm leave 🙇

**Comment by lschurr** [comment]
Still on hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
Still holding

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold.

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Still on hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by twisterdotcom** [comment]
Relabelling as Lauren is OOO now.

**Comment by CortneyOfstad** [comment]
Still on-hold 👍

**Comment by CortneyOfstad** [comment]
Still on hold — https://github.com/Expensify/App/issues/38025

**Comment by CortneyOfstad** [comment]
@lschurr is back in office, so unassigning myself. Thanks!

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Still on hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
Still on hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Asking for a retest here: https://expensify.slack.com/archives/C9YU7BX5M/p1747663418958919

**Comment by IuliiaHerets** [comment]
@lschurr QA team can't reproduce the issue

https://github.com/user-attachments/assets/21ac24be-854f-4401-a4ce-c307d0183d3a
https://github.com/user-attachments/assets/14c00f82-082b-432e-b53f-e81d9629e1d0

**Comment by lschurr** [comment]
Great! It sounds like this was fixed elsewhere: https://github.com/Expensify/App/issues/38025

Was there a PR created on this GH? Is any payment due @shubham1206agra @burczu?
